### ISSUE TITLE
FE Vereinsinfo: Austragungsorte werden mehrfach oder gar nicht angezeigt

### ISSUE THREAD

**Issue by svdoldie** [diddipoeler/sportsmanagement#1469]
In der Ansicht Vereinsinfo werden einige Austragungsorte mehrfach angezeigt, andere gar nicht.
Ein Beispiel ist hier zu sehen: https://www.svdoerpum.de/verein/clubinfo/0/18:2022-2023/104:kreispokal/13:sv-doerpum
Dort müsste auch B-Platz Dörpum auftauchen, der ist der 2. Mannschaft als Austragungsort zugeordnet.
In den Stammdaten ist beim Verein kein Austragungsort eingetragen, weil sonst nur dieser angezeigt wird. 
Hier kann man sehen, das bei einem anderen Verein auch mehrere Austragungsorte angezeigt werden:
https://www.svdoerpum.de/senioren/2-herren/clubinfo/0/0/105:kreiscup-2022-2023/17:sg-langenhorn-enge

**Comment by svdoldie** [comment]
Ja, das Problem besteht noch

**Comment by ortwin20000** [comment]
https://www.svdoerpum.de/verein/clubinfo/0/18:2022-2023/104:kreispokal/13:sv-doerpum

Bei deinem Beispiel sehe ich aber den zweiten Spielort.

**Comment by svdoldie** [comment]
Weil ich da per Eingriff in die DB nachgeholfen habe. 
Sieh dir das 2. Bespiel an, das ist es zu sehen

**Comment by ortwin20000** [comment]
Damit ich das nachstellen kann:
Ist beim Verein ein Austragungsort eingestellt?
Bei den Mannschaften des Vereins ist jeweils ein Austragungsort eingestellt?

**Comment by svdoldie** [comment]
Ja, so ist das bei mir auch. Ich weiß leider nicht mehr, wo ich das geändert habe,
damit es bei meinem Verein stimmt.

**Comment by ortwin20000** [comment]
Ich kann das bei mir nicht nachstellen. 
Es werden mir alle Spielorte einmal und nicht mehrfach des Vereins im Projekts angezeigt.
Verein hat einen Standard-Spielort eingestellt. Die Mannschaften teils unterschiedliche und gleiche. Da wird mir nur der Spielort vom Verein angezeigt. Wenn ich dann dem Projekt die Mannschaften zuweise werden mir auch die in den Mannschaften eingetragenen Spielorte einmal und nicht mehrfach angezeigt.

Es gibt hier viele Konstellationen die eingestellt werden können.
Kann du deine Einstellungen mir irgendwie (Teamviewer, Anydesk, ...) mal zeigen?

**Comment by ortwin20000** [comment]
Ich würde es trotzdem sicherstellen das es nicht ein Fehler von JSM ist. Meine E-Mail ist dir bekannt?

**Comment by svdoldie** [comment]
Ja, du hast Post 👍

**Comment by ortwin20000** [comment]
Könnten wir uns das mal gemeinsam anschauen?
Ich hab keinen Plan wie du die Sachen hier eingerichtet hast.
Wie komme ich z.B. im Frontend auf die [Vereinsseite von SG Langenhorn/Enge](https://www.svdoerpum.de/senioren/2-herren/clubinfo/0/0/105:kreiscup-2022-2023/17:sg-langenhorn-enge)?
Wenn ich den Link richtig lese wird mir vom Projekt 105 die Vereinsseite dargestellt.
Auf dieser Seite wird mir nur die Mannschaft SG Langenhorn/Enge aus dem Projekt 104 aufgelistet.
Was ist mit den anderen Mannschaften des Vereins die ebenfalls in der aktuellen Saison spielen?

Im Projekt 105 ist die Mannschaft SG Langenhorn/Enge II eingetragen. Austragungsort Waldstadion Langenhorn
Im Projekt 104 ist die Mannschaft SG Langenhorn/Enge eingetragen. Austragungsort B-Platz Enge
Im Projekt 108 ist die Mannschaft SG Langenhorn/Enge II eingetragen. Austragungsort Waldstadion Langenhorn

Sollten diese nicht auch aufgelistet werden?
Im Template Kreisliga 2016/2017 ist es zumindest so ausgewählt.
![grafik](https://github.com/diddipoeler/sportsmanagement/assets/15192502/33c7d15e-e6fb-48a7-acbf-fe85cac4cca0)

**Comment by svdoldie** [comment]
Wie gesagt, ich importiere alle Ligen aus dem DFBnet, die da hinterlegten Spielorte werden entsprechend zugeordnet. Ich habe die Anzeige der Views (z.B. in der Tabelle) für andere Vereine ausgeschaltet, da dort zum Teil ein ziemliches Durcheinander entstanden ist. Deswegen kommt man da nicht so leicht hin. Die Mannschaften müssten aber eigentlich entsprechend der Saison angezeigt werden, da könnte aber auch ein Zusammenhang mit dem Fehler #1472 sein, bzgl. der falsch gesetzten Saison-ID

**Comment by svdoldie** [comment]
Hier kann auch zugemacht werden, oder?

**Comment by ortwin20000** [comment]
Da muss ich noch drüberschauen was in der Datenbank eingetragen ist.

**Comment by svdoldie** [comment]
ok 👍

**Comment by ortwin20000** [comment]
Ich denke ich habe die Ursache gefunden:
In der Datei components\com_sportsmanagement\views\clubinfo\view.html.php ab Zeile 54:
```
		$this->teams     = sportsmanagementModelClubInfo::getTeamsByClubId($this->config['show_teams_of_club']);

		if (sportsmanagementModelClubInfo::$projectid)
		{
			$this->stadiums    = sportsmanagementModelClubInfo::getStadiums();
			$this->playgrounds = sportsmanagementModelClubInfo::getPlaygrounds();
		}
```

Hier wird in deinem [Beispiel](https://www.svdoerpum.de/senioren/2-herren/clubinfo/0/0/105:kreiscup-2022-2023/17:sg-langenhorn-enge) die Mannschaft SG Langenhorn/Enge gefunden da der Wert 2 aus `$this->config['show_teams_of_club']` übergeben wird.

Nur wird in dieser Methode `$this->stadiums    = sportsmanagementModelClubInfo::getStadiums();` ebenfalls die Methode getTeamsByClubId aufgerufen nur diesmal ohne einer Wertübergabe. Somit werden alle Mannschaften des Clubs gefunden.

Auch in der Methode `$this->playgrounds = sportsmanagementModelClubInfo::getPlaygrounds();` wird dann wiederum `sportsmanagementModelClubInfo::getStadiums();` aufgerufen.

Mein Vorschlag sieht so aus:
Änderung der beiden statischen Funktionen auf
`public static function getStadiums($show_teams_of_club = 1)`
und 
`public static function getPlaygrounds($show_teams_of_club = 1)`

In der Methode getPlaygrounds den Aufruf getStadium erweitern mit $show_teams_of_club.
Und in der view.html.php
```
			$this->stadiums    = sportsmanagementModelClubInfo::getStadiums($this->config['show_teams_of_club']);
			$this->playgrounds = sportsmanagementModelClubInfo::getPlaygrounds($this->config['show_teams_of_club']);
```

**Comment by ortwin20000** [comment]
@svdoldie Wie schaut es bei dir aus? Ich glaube wir können hier zumachen.

**Comment by svdoldie** [comment]
Nee, irgenwie stimmt da noch etwas nicht, siehe https://www.svdoerpum.de/senioren/2-herren/clubinfo/0/0/105:kreiscup-2022-2023/17:sg-langenhorn-enge
Da fehlen das Walstadion Langenhorn und das Arne-Ohlsen-Stadion in Enge.
Ist aber evtl. nur bei mir so. Dann kann ich hier zumachen

**Comment by ortwin20000** [comment]
Ich bin jetzt damit weiter und es liegt an deinen Daten die durch den DFBImport und der Neuanlage von Saisonen diese in der Tabelle `saison_team_id` eingetragen haben. Danach wurde aber im Projekt die Saison-ID geändert und die Tabelle `saison_team_id` wurde aber nicht aktualisiert.

Sieht man schön mit dieser SQL-Abfrage:
`SELECT st.id, st.season_id as season_team_pid, p.season_id as pid FROM `jsm_sportsmanagement_season_team_id` st JOIN jsm_sportsmanagement_project_team pt ON st.id=pt.team_id JOIN jsm_sportsmanagement_project p ON pt.project_id=p.id WHERE p.season_id <> st.season_id;`

Kannst du mal diese Einträge richtig stellen?
Danach sollten auch die Spielorte korrekt gefunden werden und auch alle Mannschaften des Clubs für die aktuelle Saison.

**Comment by svdoldie** [comment]
Guten Morgen :-)
Die Anfrage ergibt diese Ergebnis (ich zeige hier nur einen Teil)
Kann man die Abfrage nicht so erweitern, dass die Einträge gleich korrigiert werden?
BlueMail <https://bluemail.me> for Desktop herunterladen
Andreas Haunold wrote:
Ich bin jetzt damit weiter und es liegt an deinen Daten die durch den DFBImport und der Neuanlage von Saisonen diese in der Tabelle saison_team_id eingetragen haben. Danach wurde aber im Projekt die Saison-ID geändert und die Tabelle saison_team_id wurde aber nicht aktualisiert.
Sieht man schön mit dieser SQL-Abfrage:
SELECT st.id, st.season_id as season_team_pid, p.season_id as pid FROM jsm_sportsmanagement_season_team_id st JOIN jsm_sportsmanagement_project_team pt ON st.id=pt.team_id JOIN jsm_sportsmanagement_project p ON pt.project_id=p.id WHERE p.season_id <> st.season_id;
Kannst du mal diese Einträge richtig stellen?
Danach sollten auch die Spielorte korrekt gefunden werden und auch alle Mannschaften des Clubs für die aktuelle Saison.

**Comment by ortwin20000** [comment]
Ein einfaches Update geht da leider nicht da du bereits manche der betroffenen Teams in der Saison mit der ID 18 in der Tabelle hast.
Ein Beispiel wie du hier vorgehen könntest aber **vorher bitte ein Backup der Datenbank machen**. Keine Garantie das es so funktioniert ;-)

In der Tabelle `season_team_id` dieses Statement versuchen:
`update jsm_sportsmanagement_season_team_id st
set st.season_id = 
(SELECT p.season_id FROM jsm_sportsmanagement_season_team_id st1 JOIN jsm_sportsmanagement_project_team pt ON st1.id=pt.team_id JOIN jsm_sportsmanagement_project p ON pt.project_id=p.id WHERE st.id = pt.team_id)
where st.id in (SELECT st2.id FROM jsm_sportsmanagement_season_team_id st2 JOIN jsm_sportsmanagement_project_team pt ON st2.id=pt.team_id JOIN jsm_sportsmanagement_project p ON pt.project_id=p.id WHERE p.season_id <> st2.season_id);`

Dieser Befehl sollte die Einträge richtig stellen aber da es bereits Teams in der Saison 18 gibt wird er fehlschlagen wegen doppelter Einträge.

Der Eintrag mit der ID 1343 ist für die Saison 18 mit der ID 1463 bereits vorhanden. Hier müsstest du in der Tabelle `project_team` den Eintrag mit der team_id 1343 ändern auf 1463. Danach könntest du den Eintrag 1343 aus der Tabelle `season_team_id` entfernen.

Dann wieder das Update-Statement ausführen und beim nächsten doppelten Schlüssel die gleiche Prozedur durchführen.

**Comment by ortwin20000** [comment]
Können wir hier zumachen?
